internal/bminventory/inventory_v2_handlers.go

```diff
@@ -311,7 +316,36 @@ func (b *bareMetalInventory) V2CompleteInstallation(ctx context.Context, params
 		return common.GenerateErrorResponder(err)
 	}
 
-	if !*params.CompletionParams.IsSuccess {
+	parse := func(data interface{}) ([]models.OperatorMonitorReport, error) {
+		raw, err := json.Marshal(data)
+		if err != nil {
+			return nil, err
+		}
+		var result []models.OperatorMonitorReport
+		if err := json.Unmarshal(raw, &result); err != nil {
+			return nil, err
+		}
+		return result, nil
+	}
+
+	if !swag.BoolValue(params.CompletionParams.IsSuccess) {
+		// Check if there is a report on failing operators, and compose a report event.
+		// The event simply shows a list of failed operators' names.
+		if data, ok := params.CompletionParams.Data[clusterOperatorReportKey]; ok {
+			if ops, err := parse(data); err == nil {
+				//filter only the names of failed operators
+				failedOps := funk.Filter(ops, func(op models.OperatorMonitorReport) bool {
+					return op.Status == models.OperatorStatusFailed
+				}).([]models.OperatorMonitorReport)
+				names := funk.Map(failedOps, func(op models.OperatorMonitorReport) string {
+					return op.Name
+				}).([]string)
+				eventgen.SendClusterOperatorReportEvent(ctx, b.eventsHandler, params.ClusterID, fmt.Sprint(strings.Join(names, ", ")))
+			} else {
+				log.Errorf("Bad format of cluster operator report %s", err.Error())
```

Comment:
Since this path does not return any errors and only log a message it requires a whole new test initialization to actually test it and it is hardly worth it in this case. I did test it manually though and see this message printed as expected.

Reply:
Fair enough.
Please append the cluster and host IDs to this log line so readers would know where the error is coming from

Reply:
it happens automatically in the filters as a cross cut concern. This statement in line 310 `log = logutil.FromContext(ctx, b.log)` is wrapping the log with the relevant ids.